### PR TITLE
feat (client): react layout

### PR DESCRIPTION
**Pull request by nihalanisumit**
Basic react layout setup. Added navigation bar and modal component

## PR Checklist
Please check that your PR fulfills the following requirements:

- [X] The commit message follows [our guidelines](https://github.com/pesto-students/batch-9-Lumen/wiki/Commit-message-convention)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [X] GIFs/Videos/Images added (only for frontend changes)

## PR Type
What kind of change does this PR introduce? 

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Infrastructure changes
- [ ] Other... Please describe:
Issue Number: 
#

## Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

File: client/src/App.jsx
Comment (nimish-gupta) [inline client/src/App.jsx]: You can skip div

File: client/src/components/UI/Modal/Modal.jsx
Comment (nimish-gupta) [inline client/src/components/UI/Modal/Modal.jsx]: New line ?

File: client/src/components/UI/Modal/Modal.jsx
Comment (nimish-gupta) [inline client/src/components/UI/Modal/Modal.jsx]: You can also implement it using hooks

**Review by nimish-gupta — CHANGES_REQUESTED** [review]

**Review by piyushagwl007 — COMMENTED** [review]
Do the changes that Nimish has asked otherwise LGTM.

**Review by karansapolia — APPROVED** [review]
Nimish's comments, then LGTM

File: client/src/components/UI/Modal/Modal.jsx
Comment (nihalanisumit) [reply client/src/components/UI/Modal/Modal.jsx]: Used React.memo.

File: client/src/components/UI/Backdrop/Backdrop.jsx
Comment (nimish-gupta) [inline client/src/components/UI/Backdrop/Backdrop.jsx]: Is husky working properly? Eslint should send an error of line length

File: client/src/components/hoc/Layout/Layout.jsx
Comment (nimish-gupta) [inline client/src/components/hoc/Layout/Layout.jsx]: Can be converted to functional component

**Review by nimish-gupta — CHANGES_REQUESTED** [review]
LGTM after the changes

File: client/src/App.jsx
Comment (piyushagwl007) [inline client/src/App.jsx:31]: I would suggest creating an array containing routes and its components like:

```
[
{
 name: 'HOME',
path: '/home',
component:HOME
},
...
]
```
This will take our number of routes to another file. :smile:

File: client/src/components/Logo/Logo.jsx
Comment (piyushagwl007) [inline client/src/components/Logo/Logo.jsx:7]: I would suggest wrapping up img tag as an independent component. I am sure that will help us in future development a lot

File: client/src/App.jsx
Comment (piyushagwl007) [inline client/src/App.jsx:14]: Aren't we using Browserrouter ? or Any form of Router?

**Review by piyushagwl007 — APPROVED** [review]
LGTM. Just look into the changes that I have asked :smile:

File: client/src/components/Logo/Logo.jsx
Comment (nihalanisumit) [reply client/src/components/Logo/Logo.jsx:7]: Sure, actually img is wrapped in a Logo component but that is specific to our logo.

File: client/src/App.jsx
Comment (nihalanisumit) [reply client/src/App.jsx:14]: @piyushagwl007 yes BrowserRouter check out index.js!
I am just using route to define routes.

File: client/src/components/UI/Backdrop/Backdrop.jsx
Comment (nihalanisumit) [reply client/src/components/UI/Backdrop/Backdrop.jsx]: @nimish-gupta Actually eslint is working on staged file and I commited and pushed this first then pulled changes from husky branch.
I am going to fix all the existing lint problems in this PR now.

**Review by nimish-gupta — APPROVED** [review]

**Pull request merged**
